syntax = "proto2";

package android.vts;

message VTableEntry {
  optional int64 offset = 1;
  enum Kind {
    UNDEFINED = 0;
    VCALLOFFSET = 1;
    VBASEOFFSET = 2;
    OFFSETTOTOP = 3;
    RTTI = 4;
    VFUNCPOINTER = 5;
    DELETINGDTORPOINTER = 6;
    COMPLETEDTORPOINTER = 7;
  }
  optional Kind kind = 2 [default = UNDEFINED];
  optional string name = 3;
  optional string demangled_name = 4;
  optional bool is_pure = 5 [default = false];
}

message VTable {
  optional string name = 1;
  optional string demangled_name = 2;
  repeated VTableEntry vtable_entries = 3;
}

message Symbol {
  enum Kind {
    UNDEFINED = 0;
    GLOBAL = 1;
    WEAK = 2;
  }
  optional Kind kind = 1 [default = UNDEFINED];
  optional string name = 2;
}

message AbiDump {
  repeated VTable vtables = 1;
  repeated Symbol symbols = 2;
}
